ocs_ci/templates/mcg/java/s3test/src/main/java/amazons3/s3test/ChunkedUploadApplication.java

```diff
@@ -0,0 +1,231 @@
+import com.amazonaws.ClientConfiguration;
+import com.amazonaws.SDKGlobalConfiguration;
+import com.amazonaws.SdkClientException;
+import com.amazonaws.auth.AWSStaticCredentialsProvider;
+import com.amazonaws.auth.BasicAWSCredentials;
+import com.amazonaws.client.builder.AwsClientBuilder.EndpointConfiguration;
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.AmazonS3ClientBuilder;
+import com.amazonaws.services.s3.model.CompleteMultipartUploadRequest;
+import com.amazonaws.services.s3.model.GetObjectRequest;
+import com.amazonaws.services.s3.model.InitiateMultipartUploadRequest;
+import com.amazonaws.services.s3.model.InitiateMultipartUploadResult;
+import com.amazonaws.services.s3.model.PartETag;
+import com.amazonaws.services.s3.model.PutObjectRequest;
+import com.amazonaws.services.s3.model.UploadPartRequest;
+import com.amazonaws.services.s3.model.UploadPartResult;
+
+import software.amazon.awssdk.utils.Pair;
+
+/**
+ * Hello world!
+ *
+ */
+//public class App
+//{
+//    public static void main( String[] args )
+//    {
+//        System.out.println( "Hello World!" );
+//    }
+//}
+
+public class ChunkedUploadApplication {
+
+	public static void main(String[] args) throws NoSuchAlgorithmException, IOException {
+		System.setProperty(SDKGlobalConfiguration.DISABLE_CERT_CHECKING_SYSTEM_PROPERTY, "true");
+
+		System.out.println("Endpoint: "+ args[0]);
+		System.out.println("Access Key: "+ args[1]);
+		System.out.println("Secret Key: "+ args[2]);
+		System.out.println("bucket name: "+ args[3]);
+		System.out.println("Is multipart?: "+ args[4]);
+
+		String endpoint = args[0];
+		String ACCESS_KEY = args[1];
+		String SECRET_KEY = args[2];
+		String bucketName = args[3];
+		Boolean isMultipart = false;
+
+		if (args[4] == "True")
+			isMultipart = true;
+		AmazonS3 s3Client = configureS3Client(endpoint, ACCESS_KEY, SECRET_KEY);
+		Boolean test1 = testCycle(s3Client, bucketName, isMultipart); // regular bucket + regular upload
+		int success = 1;
+		if (test1 == true) {
+			System.out.println("TEST FINISHED SUCCEFULLY!!");
+			success = 0;
+		}
+		else {
+			System.out.println("TEST FAILED!!");
+		}
+		System.exit(success);
+		return;
+	}
+
+	public static Boolean testCycle(AmazonS3 s3Client, String bucketName, Boolean multipartUpload)
+			throws IOException, NoSuchAlgorithmException {
+
+		String OriginalFilePath = "TestFile" + Instant.now().toString();
+		String localFilePath = "PATH2";
+		File localFile = new File(localFilePath); // creates an empty file for the output
+
+		try {
+			// creates a file of size 8MB contains random content.
+			File originalFile = createRandomContentFile(OriginalFilePath, 15 * 1024 * 1024);
+			Boolean uploadSuceeded = multipartUpload
+					? multiPartUpload(s3Client, bucketName, OriginalFilePath, originalFile)
+					: regularUpload(s3Client, bucketName, OriginalFilePath, originalFile);
+
+			Boolean downloadSuceeded = download(s3Client, bucketName, OriginalFilePath, localFile);
```

Comment:
If upload fails (line 90), then you can return false and don't have to try the download & compare because you don't return the success/failure separately.
IMO, you should move the conditional "ifs" after every operation and return if it failed.